### PR TITLE
✨ Allow config difference core trend timeframe

### LINKED ISSUE #44: ✨ `Multi-TimeFrame` usage + Automated zooming
## The idea here is to:
- Make the buy/sell signals more robust due to more confirmation over more different candles
- Trade safer overall by doing more swing trades (Bigger candles give more certain predictions since they aren't so clouded by volatility)
- Make more profit, by going more and more near scalping mode when the market goes sideways (by detecting smaller trends and doing smaller shorter trades for smaller profits)

## How this would be implemented:
- Use multiple candle data like we're doing now with [TimeFrame-Zoom](https://github.com/Rikj000/MoniGoMani/blob/main/MGM_DOCUMENTATION.md#timeframe-zoom) (TFZ), but even more:
   - 5m for original intra candle fluctuations during testing (purpose of TFZ)
   - Then for normal candle usage I was thinking about for example using:
      - 15m + 30m + 1h +  2h candles in a similar fashion as we combine candles in the dataframe now (so base timeframe will be 5m but we'll add the other data into it, this will make it rather heavy though), but being frugal in the amount of candles we'll use to not overload systems.

- We'd use all that candle data to detect which candles we'd like to use:
   - Base is 2hr, if this has an upwards/downwards trend detected then we use the buy/sell signals for it (upwards/downwards is currently enabled in the config)
   - If we use the signals from the 2hr timeframe, we zoom in, check if there the same trend is also detected, if so, add the buy/sell signals (2 candle sizes now confirm that it's a good moment to buy/sell)
   - We continue doing this till we don't detect a matching trend anymore (ideally we'd zoom in all the way to 5m, then all candle sizes give indication that it's a good time to buy/sell)

- If we detect sideways movement on 2hrs then we don't use that candle data for our buy/sell signals, but then:
   - We zoom in to 1hrs, to see if we can detect a trend here, if this is the case we start using the signals as buy/sell signals
   - We still continue zooming in if still no trend would have been found on the 1hrs

### PR DESCRIPTION
**Pull request by taina0407**
- Add new 'trend_timeframe' in mgm-config.json to setting timeframe to detect core trend, if not set then using default timeframe

# TODO: 
- Didn't found any documentation about HT_TRENDMODE required startup candles number, so currently setting it to 200. This should to optimize incase of using long core trend timeframe like 1d might cause the bot loading candle pretty slow
- Using better logic to merge core_trend column to informative and main dataframe, because I'm quite new on python so don't know a good way to implement it yet, currently using manual rename and delete

### CODE REVIEW

File: user_data/strategies/MasterMoniGoManiHyperStrategy.py [hunk 1]
```diff
@@ -457,6 +459,15 @@ def _populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFram
         """
 
         tfz = 'TimeFrame-Zoom'
+        # Populate core trend indicators
+        core_trend = load_pair_history(pair=metadata['pair'],
+                                       datadir=self.config['datadir'],
+                                       timeframe=self.core_trend_timeframe,
+                                       # ToDo: calculate correct startup_candles needed for HT_TRENDMODE and SAR
+                                       startup_candles=self.startup_candle_count,
```
Comment (Rikj000) [inline user_data/strategies/MasterMoniGoManiHyperStrategy.py:467]: For now I think it's fine to set the `startup_candle_count` to the one defined in `mgm-config`

File: user_data/strategies/MasterMoniGoManiHyperStrategy.py [hunk 2]
```diff
@@ -473,8 +484,19 @@ def _populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFram
...
             first_informative = dataframe['date'].min().floor('H')
             informative = informative[informative['date'] >= first_informative]
-            # Populate core trend indicators
-            informative = self._populate_core_trend(informative, metadata)
+
+            # Merge core trend to informative data frame
+            informative = merge_informative_pair(
+                informative, core_trend[['date', 'ht_trendmode', 'sar', 'trend']].copy(),
+                self.informative_timeframe, self.core_trend_timeframe, ffill=True)
+            informative['ht_trendmode'] = informative[f'ht_trendmode_{self.core_trend_timeframe}']
```
Comment (Rikj000) [inline user_data/strategies/MasterMoniGoManiHyperStrategy.py]: Indeed, there must be a cleaner way to do this :slightly_smiling_face:

**Review by Rikj000 — CHANGES_REQUESTED** [review]
Did some improvements to the code already as I was setting up a development environment for this PR.
I'll get to more thorough debugging of the code soon & will see if all checks out and if I can find a way to do this with a bit less code duplication.

@taina0407 already thank you for this PR! :pray:
Feel free to also try to find a solution for this with less code duplication :slightly_smiling_face:

File: user_data/strategies/MasterMoniGoManiHyperStrategy.py (hunk 2 above)
Comment (taina0407) [reply user_data/strategies/MasterMoniGoManiHyperStrategy.py]: Updated rename logic using panda rename.
Also tried to debug using PyCharm and verified that the core trend is set correctly in both backtest and dryrun mode 🥰

**Review by Rikj000 — APPROVED** [review]
Also tested with PyCharms debugger and the implementation looks good to me! :smile: 

I took the liberty of also removing some double/unused dataframe columns which I noticed during my debugging, this should result in less memory usage & perhaps a faster HyperOpt too :slightly_smiling_face:

However I still need to thoroughly test both of our changes now before I get to merging.
Feel free to already append some test results:
- From before the PR implementation as comparison
- Some of the best results yielded with the new feature implemented

So we can be sure & document it in this PR that it does indeed bring improvements

**Comment by Rikj000** [comment]
With the timeframes configured as following I was already able to yield better results on one of my first tests:
```json
{
    "timeframe": "30m",
    "trend_timeframe": "4h",
    "backtest_timeframe": "5m"
}
```
core-trend-4h

**Comment by Rikj000** [comment]
@taina0407 `1d` didn't appear to yield me better results, so I guess I'll make the default `4h`.
Then it's up to users to test & improve further upon this settings :slightly_smiling_face: 
```json
{
    "timeframe": "30m",
    "trend_timeframe": "1d",
    "backtest_timeframe": "5m"
}
```
1d-core-trend
However as discussed in PMs the implementation of this setting likely will still change a bit with the implementation of [issue 44](https://github.com/Rikj000/MoniGoMani/issues/44)

**Pull request merged**